homeassistant/components/yalexs_ble/util.py

```diff
@@ -0,0 +1,58 @@
+"""The yalexs_ble integration models."""
+from __future__ import annotations
+
+from yalexs_ble import local_name_is_unique
+
+from homeassistant.components.bluetooth import (
+    BluetoothScanningMode,
+    BluetoothServiceInfoBleak,
+    async_discovered_service_info,
+    async_process_advertisements,
+)
+from homeassistant.components.bluetooth.match import (
+    ADDRESS,
+    LOCAL_NAME,
+    BluetoothCallbackMatcher,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DEVICE_TIMEOUT
+
+
+def bluetooth_callback_matcher(
+    local_name: str, address: str
+) -> BluetoothCallbackMatcher:
+    """Return a BluetoothCallbackMatcher for the given local_name and address."""
+    if local_name_is_unique(local_name):
+        return BluetoothCallbackMatcher({LOCAL_NAME: local_name})
```

Comment:
why do we prefer to match on name over address?

Reply:
Since macOS hides the address with a uuid that can change if you restore the config on another instance, the local name is a bit more stable when we know it's unique since it's the serial number on all the newer locks